lms/djangoapps/certificates/migrations/0037_fix_legacy_broken_invalid_certs.py

```diff
@@ -0,0 +1,28 @@
+# Generated by Django 3.2.23 on 2024-01-25 21:56
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    """
+    If any certificates exist with an invalidation record that are not marked as unavailable,
+    change their status. Irreversible.
+    """
+
+    dependencies = [
+        ("certificates", "0036_modifiedcertificatetemplatecommandconfiguration"),
+    ]
+
+    def make_invalid_certificates_unavailable(apps, schema_editor):
+        GeneratedCertificate = apps.get_model("certificates", "GeneratedCertificate")
+
+        GeneratedCertificate.objects.filter(
+            certificateinvalidation__active=True
+        ).exclude(status="unavailable").update(status="unavailable")
```

Comment:
I would love a pretty careful read from code review here, even though this is a very simple query, because I forced a bunch of certificates in my local development environment into the broken state but that is obviously inherently not the same thing as the real ones.